CMakeLists.txt

```diff
@@ -37,8 +37,8 @@ if(CMAKE_BUILD_TYPE STREQUAL "PyPI")
     set(GYMPP_BUILD_SHARED_LIBRARY FALSE
         CACHE BOOL "Compile libraries as shared libraries" FORCE)
 else()
-    set(GYMPP_BUILD_SHARED_LIBRARY TRUE BOOL
-        CACHE BOOL "Compile libraries as shared libraries")
+    set(GYMPP_BUILD_SHARED_LIBRARY TRUE
+        CACHE BOOL "Compile libraries as shared libraries" FORCE)
```

Comment:
What is the point of having a user-facing option if you always force it to a value?

Reply:
Mmh the `FORCE` option should not be there, good catch. I remember why I used it in the other conditional branch, and using it also here as you said is not useful. I think I should come up with a better logic that sets static compilation in PyPI build mode without overriding the cache value. This could be useful to switch back to e.g. Release and not loosing the preferred build type.

Reply:
> I think I should come up with a better logic that sets static compilation in PyPI build mode without overriding the cache value. This could be useful to switch back to e.g. Release and not losing the preferred build type.

In the case of the `CMAKE_BUILD_TYPE` equal to `PyPI` , you could just set directly `BUILD_SHARED_LIBS` ignoring `GYMPP_BUILD_SHARED_LIBRARY` . Note that this could create problems if `gym-ignition` is included in a CMake project that defines `BUILD_SHARED_LIBS` as an option, but I think that this workflow combine with `CMAKE_BUILD_TYPE` equal to `PyPI`  is  not a configuration that it make sense to support.

Reply:
Implemented in https://github.com/robotology/gym-ignition/compare/f9305db800b386d5cd1fc2c9764f84f5273d5f4a..a6a343282980c5f195f27c4ea89f0d1fd111e5e9

Reply:
> I think that this workflow combine with CMAKE_BUILD_TYPE equal to PyPI is not a configuration that it make sense to support.

In general, users should not even know that the `PyPI` build type exists. It is there only to simplify the CD pipeline.